pkg/systemaccount/systemaccount.go

```diff
@@ -116,28 +119,32 @@ func (s *Manager) GetOrCreateSystemClusterToken(clusterName string) (string, err
 func (s *Manager) GetOrCreateProjectSystemAccount(projectID string) error {
 	_, projectName := ref.Parse(projectID)
 
-	user, err := s.GetProjectSystemUser(projectName)
+	u, err := s.GetProjectSystemUser(projectName)
 	if err != nil {
 		return err
 	}
 
-	bindingName := user.Name + "-member"
-	_, err = s.prtbs.GetNamespaced(projectName, bindingName, v1.GetOptions{})
-	if err == nil {
-		return nil
-	} else if errors2.IsNotFound(err) {
-		_, err = s.prtbs.Create(&v3.ProjectRoleTemplateBinding{
+	bindingName := u.Name + "-member"
+	_, err = s.prtbLister.Get(projectName, bindingName)
+	if err != nil {
+		if !errors2.IsNotFound(err) {
+			return err
+		}
+		// prtb does not exist in cache, attempt to create it
+		prtb := &v3.ProjectRoleTemplateBinding{
 			ObjectMeta: v1.ObjectMeta{
 				Name:      bindingName,
 				Namespace: projectName,
 			},
 			ProjectName:      projectID,
-			UserName:         user.Name,
+			UserName:         u.Name,
 			RoleTemplateName: projectMemberRole,
-		})
+		}
+		if _, err := s.prtbs.Create(prtb); err != nil && !errors2.IsAlreadyExists(err) {
```

Comment:
Why would we need to add this alreadyExists check if we just found that prtb doesn't exist, is it for HA or something ?

Reply:
It does't exist in the cache, a cache miss will still cause an error here if it already exists in the DB.

Reply:
In general create call errors should get protected with an IsAlreadyExists check.